sklearn/model_selection/_search.py

```diff
@@ -379,6 +381,58 @@ def check(self):
     return check
 
 
+def _yield_masked_array_for_each_param(
+    candidate_params: Sequence[dict[str, Any]],
+) -> Iterator[tuple[str, MaskedArray]]:
+    """
```

Comment:
We've decided to not add type hints so far. Since the input and output types of this function are quite simple I'd not add them here.